src/recipes.rs

```diff
@@ -0,0 +1,50 @@
+// use crate::api::get_user_data;
+use crate::api::get_user_data;
+use crate::protobuf::anylist::{PbIngredient, PbRecipe as ApiRecipe, PbRecipeDataResponse};
+use std::error::Error;
+
+pub struct Ingredient {
+    pub name: String,
+    pub quantity: f32,
+    pub unit: String,
+}
+#[derive(Debug, Clone)]
+pub struct Recipe {
+    pub id: String,
+    pub name: String,
+    // pub ingredients: Vec<Ingredient>,
+    // pub preparation_steps: Vec<String>,
+    pub source_url: String,
+}
+
+pub async fn get_recipes(signed_user_id: &str) -> Result<Vec<Recipe>, Box<dyn Error>> {
+    let data = get_user_data(signed_user_id).await?;
+    let recipes = match data.recipe_data_response {
+        Some(ref res) => recipes_from_response(res.clone()),
+        None => Vec::new(),
+    };
+    Ok(recipes)
+}
+
+fn recipes_from_response(response: PbRecipeDataResponse) -> Vec<Recipe> {
+    let mut recipes: Vec<Recipe> = Vec::new();
```

Comment:
Personally, I would consider having this as an `impl` on Recipe, like

```rust
impl Recipe {
    fn from_recipe_data_response(response: PbRecipeDataResponse) -> Vec<Recipe> {
        // ...
    }
}
```

but it doesn't make a ton of sense, I'm not particularly experienced at Rust, and what you have definitely works!